### ISSUE TITLE
Feature request: support zoxide

### ISSUE THREAD

**Issue by Seirdy** [changyuheng/fz#24]
Currently, fz supports [rupa/z](https://github.com/rupa/z) and [z.lua](https://github.com/skywind3000/z.lua). This is quite nice because it gets both of these to behave in (mostly) the same way. It would be great if it could also support [zoxide](https://github.com/ajeetdsouza/zoxide); that way, migrating between implementations could be even easier.

**Comment by ajeetdsouza** [comment]
zoxide v0.8.0 now includes completions of its own.

@Seirdy perhaps this can be closed? If there's anything specific you want from the new completions, you can request it there itself.

**Comment by johannjhang** [comment]
zoxide looks pretty promising. I think I can archive this repo now.

**Comment by johannjhang** [comment]
Just gave zoxide a try, it'ss not as solid as z at the moment. Reopened this project as I'm going back for it.

**Comment by ajeetdsouza** [comment]
Hey @changyuheng, I'd love to hear your feedback. What did you think was missing / not as good?

**Comment by johannjhang** [comment]
Hi @ajeetdsouza, sorry, I need to amend what I said. It's solid, I just couldn't use it in one of my testing environments. I realized that it's because I didn't install it in that environment later. But one thing I don't like is the tab completion behavior, and it was the main reason I reopened this project. I don't like to type another space just for being able to launch the tab completion. All the tab completion in other commands even in text editors don't work in that way. So it really breaks the united user experience.

**Comment by ajeetdsouza** [comment]
That's fair enough. The reason the space is necessary is because the `z` command is overloaded as both a `cd` as well as a `jump` command, so there needs to be something to distinguish between the two types of completions.

The idea is not necessarily unique -- it was inspired by fzf's own [completion scripts](https://github.com/junegunn/fzf/#fuzzy-completion-for-bash-and-zsh). fzf uses `**<TAB>`, but for zoxide I made it `<SPACE><TAB>` since that's much easier to type.

**Comment by Seirdy** [comment]
I'd also prefer not to archive this project because I've found that z.lua is as fast zoxide if I build luajit--or even lua--as a statically-linked binary with lots of compiler optimization flags and some PGO.

Furthermore, I'd imagine that Lua has better portability than zoxide. This is important because z-like functionality is definitely something I'd want to bring with me wherever I go: I already use it on 32bit ARMv7, x86_64, and armv8 computers running Linux and OpenBSD.

So as long as z.lua and z.sh have good use-cases, fz has reason to exist.

As for this specific issue, I think that it's certainly *less important* given that zoxide has its own completions; however, the fact that its completions are different means that the issue still has reason to exist. The advantage of fz isn't just providing completions, but making the z command work in a similar way regardless of the "backend" used.

In conclusion: z.lua has a valid use-case even though zoxide exists. This issue is relevant though less important if zoxide has its own different completions. Whether it's important enough to work on in this repository is something for @changyuheng to decide for himself.

-- 
/Seirdy

**Comment by ajeetdsouza** [comment]
@Seirdy I don't see why zoxide shouldn't support a platform that Lua supports. If you do require it on a platform that it's not available on, do file an issue; I'll add it to the list of targets in the next release.

**Comment by johannjhang** [comment]
@ajeetdsouza I just recalled the issue that made me say "zoxide not as solid as z". I'm using Zsh in MSYS2 on Windows. When I `cd` into a network drive like //a.network-drive/a-folder, zoxide will give me:

```
zoxide: could not get drive letter: \\a.network-drive\a-folder
```

**Comment by ajeetdsouza** [comment]
That's definitely a bug, thanks for reporting it! I've created an issue for it here: https://github.com/ajeetdsouza/zoxide/issues/331
